cavern/src/main/webapp/WEB-INF/web.xml

```diff
@@ -120,6 +120,35 @@
     </init-param>
     <load-on-startup>2</load-on-startup>
   </servlet>
+
+  <servlet>
+    <servlet-name>AuthFilesServlet</servlet-name>
+    <servlet-class>ca.nrc.cadc.rest.RestServlet</servlet-class>
+    <init-param>
+      <param-name>get</param-name>
+      <param-value>org.opencadc.cavern.files.GetAction</param-value>
+    </init-param>
+    <init-param>
+      <param-name>put</param-name>
+      <param-value>org.opencadc.cavern.files.PutAction</param-value>
+    </init-param>
+    <init-param>
+      <param-name>isPreAuthRequest</param-name>
+      <param-value>true</param-value>
```

Comment:
I would use different classes (that respond to different endpoints /preauth and /files) rather than in here.